### ISSUE TITLE
cache file for download?

### ISSUE THREAD

**Issue by bradfordcondon** [tripal/tripal_analysis_expression#222]
FIle generation is real slow IE here

https://www.hardwoodgenomics.org/Gene-expression-profiling/1962969?tripal_pane=group_summary_tripalpane

![screen shot 2018-09-12 at 9 11 38 am](https://user-images.githubusercontent.com/7063154/45427187-42e7a280-b66c-11e8-9ff9-48e8b4d8338e.png)

Can we work a way to cache the file created?

The main difficulty is that it relies on an mview, so it would need to be re-generated when the mview is re-populated.  So I can imagine, for example, that the field's load creates and caches the file and then the admin would just need to clear cache after popping the mview. 

 Unfortunately this would mean significantly slowing down after clearing cache, so maybe theres a better way?

**Comment by bradfordcondon** [comment]
ok, when the file is UPLOADED with the importer, just store the uploaded file somewhere.  The file location is added to the analysis as an analysisprop.  Then the field uses that prop to point to hte file for download.

**Comment by mestato** [comment]
I'd like to see us offer both the raw and normalized counts as downloadable files (but continue with storing normalized only in chado).

**Comment by mestato** [comment]
I can put paths to the htseq folders with raw counts per library on the staton server. We can host them like that or someone will need to turn them into a matrix file.

Tulip poplar:

/staton/projects/hardwoods/unigenes/1.1_tulip_poplar/version_10_13_2014/12_htseq

Green ash:

/staton/projects/hardwoods/unigenes/1.4_green_ash/version_120313_miseq_trinity_cdhit/7_htseq

For combined across same library sequenced multiple times, see 

/staton/projects/hardwoods/unigenes/1.4_green_ash/version_120313_miseq_trinity_cdhit/7_htseq/combined

Black walnut:

/staton/projects/hardwoods/unigenes/1.5_black_walnut/version_120313_miseq_trinity_cdhit/7_htseq

For combined across same library sequenced multiple times, see 

/staton/projects/hardwoods/unigenes/1.5_black_walnut/version_120313_miseq_trinity_cdhit/7_htseq/merged_count_files

Etc. Matt can help find the rest.

**Comment by bradfordcondon** [comment]
why would file generation be impossibly slow for a single feature?  its not that much data.  My money is its a permission problem?

**Comment by bradfordcondon** [comment]
ok.  I'll set up the file field for the analysis then.

**Comment by bradfordcondon** [comment]
<img width="1389" alt="screen shot 2018-11-05 at 11 35 50 am" src="https://user-images.githubusercontent.com/7063154/48012051-222d4d80-e0ef-11e8-9455-f4716ebd4ae3.png">

I just disabled the old field.  We could instead keep the text listing the info but remove the "click here to download" link.

What do we think? This on dev here (with a dummy file):

https://hardwoods.ag.utk.edu/Gene-expression-profiling/1962969?tripal_pane=group_file_download

**Comment by bradfordcondon** [comment]
pinging @mestato for visibility

**Comment by almasaeed2010** [comment]
When we say create a matrix file, what would that look like? Is it simply all the `filtered` files combined into 1?

**Comment by almasaeed2010** [comment]
test

**Comment by bradfordcondon** [comment]
.filtered files?

the module actually has example files for boht formates packaegd with it.

https://github.com/tripal/tripal_analysis_expression/tree/master/example_files

**Comment by almasaeed2010** [comment]
`.filtered` files on the staton server:

```bash
[Newton:staton 12_htseq]$ ls
14DAy-125ppb_S15.counts.txt           29Day-225_S16.counts.txt
14DAy-125ppb_S15.counts.txt.filtered  29Day-225_S16.counts.txt.filtered
...
```

Sample content of a file:

```bash
[Newton:staton 12_htseq]$ head 29Day-125_S15.counts.txt.filtered 
Liriodendron_tulipifera_10132014_comp0_c0_seq1	0
Liriodendron_tulipifera_10132014_comp10000_c0_seq1	0
Liriodendron_tulipifera_10132014_comp10000_c0_seq2	4
Liriodendron_tulipifera_10132014_comp10002_c0_seq1	1023
Liriodendron_tulipifera_10132014_comp10003_c0_seq1	31
Liriodendron_tulipifera_10132014_comp10004_c0_seq1	23
Liriodendron_tulipifera_10132014_comp10005_c0_seq1	1
Liriodendron_tulipifera_10132014_comp10007_c0_seq1	0
Liriodendron_tulipifera_10132014_comp10008_c0_seq1	45
Liriodendron_tulipifera_10132014_comp1000_c0_seq1	1
```

**Comment by bradfordcondon** [comment]
ok, so thats the 2 column, 1 biomaterial/file.

matrix is same rows (gene, value) but then a column for each biomaterial.

**Comment by almasaeed2010** [comment]
Script to convert the files into a matrix: https://gist.github.com/almasaeed2010/b1fa99a68c8dbcd171384a9f5611e063


Usage example:

```sh
python biomaterials_to_matrix.py *.filtered > matrix.txt
```

**Comment by jwest60** [comment]
fixed on #264
